cli/client.go

```diff
@@ -372,6 +372,34 @@ func (c *viamClient) robotsPartLogsAction(cCtx *cli.Context) error {
 	)
 }
 
+// RobotsPartRestartAction is the corresponding Action for 'machines part restart'.
+func RobotsPartRestartAction(c *cli.Context) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	return client.robotPartRestart(c)
+}
+
+func (c *viamClient) robotPartRestart(cCtx *cli.Context) error {
+	orgStr := cCtx.String(organizationFlag)
+	locStr := cCtx.String(locationFlag)
+	robotStr := cCtx.String(machineFlag)
+	partStr := cCtx.String(partFlag)
+
+	part, err := c.robotPart(orgStr, locStr, robotStr, partStr)
+	if err != nil {
+		return err
+	}
+	_, err = c.client.MarkPartForRestart(cCtx.Context, &apppb.MarkPartForRestartRequest{PartId: part.Id})
```

Comment:
Didn't add a test because I don't see an obvious value to testing this func. If the proto changes, then this will fail typechecking. If the proto doesn't change, but behavior changes, then there is no good way to unit-test this.

If you disagree, happy to wrap this with injector / a unit test.

Reply:
> there is no good way to unit-test this.

I also can't think of a good way to test this; wrapping with an injector seems like sort a straw-man kind of test.